draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -604,10 +610,19 @@ to a stub resolver for an SVCB record query:
    the results in the Additional section.
 
 3. Otherwise, select an AliasForm record at random, and resolve A, AAAA,
-   and SVCB records for
-   the SvcDomainName.  If the SVCB record does not exist, add the A and AAAA
-   records to the Additional section.  Otherwise, go to step 1,
-   subject to loop detection heuristics.
+   and SVCB records for the SvcDomainName.  If the SVCB record does not exist,
+   add all records from the A and AAAA responses to the Additional section.
+   Otherwise, go to step 1, subject to loop detection heuristics.
+
+In steps 2 and 3, "resolve" means the resolver's ordinary iterative
+resolution procedure, as if processing a query for that RRSet.
+This includes following any aliases that the resolver would ordinarily
+follow (e.g. CNAME, {{!DNAME}}).  If a queried record does not exist, the
+resolver SHOULD add to the Additional section any records that it would
+normally include in a negative response (e.g. CNAME, SOA), in order to
+enable negative caching by the stub.
+
```

Comment:
Is the negative behavior actually possible?    (Is it possible to add an empty record additionals?  I'm forgetting whether this is valid, but I can't recall seeing it done?  It may be that neither is possible?)    Also, is there a distinction to be made here between NODATA/NOERROR (the RRType doesn't exist) and NXDOMAIN?  Certainly, I don't believe the latter can be added to the Additional section?

Reply:
> Is the negative behavior actually possible?

I think it might be possible, but it's definitely not straightforward.  It would require the client to recapitulate the recursive resolution procedure, notice the absence of certain RRSets, and check for a corresponding SOA instead.  It's more obviously possible (but even more complicated) for signed zones.

Anyway, I've significantly reworked this section to try to pin down the precise behavior of which records go where, what RCODE to return, etc.